packages/datadog-plugin-couchbase/test/index.spec.js

```diff
@@ -5,297 +5,304 @@ const semver = require('semver')
-                expect(span).to.have.property('name', 'couchbase.query')
-                expect(span).to.have.property('service', 'test-couchbase')
-                expect(span).to.have.property('resource', query)
-                expect(span).to.have.property('type', 'sql')
-                expect(span.meta).to.have.property('span.kind', 'client')
-                expect(span.meta).to.have.property('couchbase.bucket.name', 'datadog-test')
+          describe('queries on buckets', () => {
+            it('should handle N1QL queries', done => {
+              const query = 'SELECT 1+2'
+
+              agent
+                .use(traces => {
+                  const span = traces[0][0]
+                  expect(span).to.have.property('name', 'couchbase.query')
+                  expect(span).to.have.property('service', 'test-couchbase')
+                  expect(span).to.have.property('resource', query)
+                  expect(span).to.have.property('type', 'sql')
+                  expect(span.meta).to.have.property('span.kind', 'client')
+                  expect(span.meta).to.have.property('couchbase.bucket.name', 'datadog-test')
+                })
+                .then(done)
+                .catch(done)
+
+              const n1qlQuery = N1qlQuery.fromString(query)
+              bucket.query(n1qlQuery, (err) => {
+                if (err) done(err)
               })
-              .then(done)
-              .catch(done)
-
-            const n1qlQuery = N1qlQuery.fromString(query)
-            bucket.query(n1qlQuery, (err) => {
-              if (err) done(err)
             })
           })
         })
       })
-    })
-
-    withVersions('couchbase', 'couchbase', '>=3.0.0', version => {
-      beforeEach(() => {
-        tracer = global.tracer = require('../../dd-trace')
-      })
-
-      describe('without configuration', () => {
-        beforeEach(done => {
-          agent.load('couchbase').then(() => {
-            couchbase = proxyquire(`../../../versions/couchbase@${version}`, {}).get()
-            couchbase.connect('couchbase://localhost', {
-              username: 'Administrator',
-              password: 'password'
-            }).then(_cluster => {
-              cluster = _cluster
-              bucket = cluster.bucket('datadog-test')
-              collection = bucket.defaultCollection()
-            }).then(done).catch(done)
-          })
-        })
-
-        afterEach(async () => {
-          await cluster.close()
-        })
-
-        after(() => {
-          return agent.close({ ritmReset: false })
-        })
-
-        it('should run the Query callback in the parent context', done => {
-          const query = 'SELECT 1+1'
-          const span = tracer.startSpan('test.query.cb')
+    } else {
+      const ignoreVersions = version =>
+        semver.intersects(version, '3.2.0 - 3.2.1') || // related to JSCBC-916
+        semver.intersects(version, '^4.1.0') // multiple spdlog error
```

Comment:
Could we ignore 4.0.0 instead of ignoring the latest version? Otherwise we won't know if something breaks with a new release of the library.

Reply:
Yeah I'll give that a shot, but I would probably have to take more of a look into why there is some multiple loggers exception. This happens when testing any multiple versions of `4.x` together (so, if I tested `^4.1.0`, it would test `4.1.0` and whatever latest `4.x` version there is, triggering the exception. I used `4.0.0` as a single version to avoid this, but also had the same thought you brought up before pushing this change up but thought I'd see what you'd say, so I'll look into it!).